tests/integration-tests/cfn_stacks_factory.py

```diff
@@ -167,21 +179,23 @@ def create_stack(self, stack):
     def delete_stack(self, name, region):
         """Destroy a created cfn stack."""
         with aws_credential_provider(region, self.__credentials):
-            id = self.__get_stack_internal_id(name, region)
-            if id in self.__created_stacks:
+            internal_id = self.__get_stack_internal_id(name, region)
+            if internal_id in self.__created_stacks:
                 logging.info("Destroying stack {0} in region {1}".format(name, region))
                 try:
-                    stack = self.__created_stacks[id]
+                    stack = self.__created_stacks[internal_id]
                     cfn_client = boto3.client("cloudformation", region_name=stack.region)
                     cfn_client.delete_stack(StackName=stack.name)
                     final_status = self.__wait_for_stack_deletion(stack.cfn_stack_id, cfn_client)
-                    self.__assert_stack_status(final_status, "DELETE_COMPLETE")
                 except Exception as e:
                     logging.error(
                         "Deletion of stack {0} in region {1} failed with exception: {2}".format(name, region, e)
                     )
                     raise
-                del self.__created_stacks[id]
+                self.__assert_stack_status(
```

Comment:
As we discuss, we can move this back to the `try` block

Reply:
Done.